### PR TITLE
Temp/brand

### PR DESCRIPTION
**Pull request by rinkishivani**

### CODE REVIEW

File: products/forms.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        widget=forms.TextInput(
+            attrs={
+                'placeholder': 'Brand Name',
+                'class': 'form-control'
+            }
+        )
+
+    )
+
+    image = forms.ImageField(
```
Comment (snehals21) [inline products/forms.py:42]: @rinkishivani why do we need max_length in ImageField? kindly give the specific field wise validation messages as here you have used same message for name and image.

cc: @lalitvasoya

File: products/views.py [hunk 2]
```diff
@@ -29,3 +36,111 @@ def home_page(request):
...
+    if request.method == "POST":
+        brand_form = UpdateBrandForm(
+            request.POST, request.FILES, instance=get_object_or_404(Brand, id=pk)
+        )
+        if brand_form.is_valid():
+            brand_form.save()
+            messages.success(request, BrandFormSuccessMessages.BRAND_UPDATED)
+        return redirect('view-brand')
+
+    brand_form = UpdateBrandForm(instance=Brand.objects.get(id=pk))
```
Comment (snehals21) [inline products/views.py]: @rinkishivani there is no need to do multiple query for Brand instance. you can query one time and assign it to a variable and then use that variable in `UpdateBrandForm(instance=Brand.objects.get(id=pk))`

cc: @lalitvasoya

File: products/views.py [hunk 3]
```diff
@@ -29,3 +36,111 @@ def home_page(request):
...
+
+    if request.method == "POST":
+        try:
+            if not product_has_brand:
+                brand.delete()
+            else:
+                raise CannotDeleteBrand
+        except CannotDeleteBrand:
+            messages.info(request, BrandFormErrorMessages.BRAND_PROTECTED)
+            breakpoint()
```
Comment (snehals21) [inline products/views.py]: @rinkishivani remove breakpoint() 

cc: @lalitvasoya

File: products/forms.py (hunk 1 above)
Comment (rinkishivani) [reply products/forms.py:42]: removed unnecessary validation.

File: products/views.py (hunk 2 above)
Comment (rinkishivani) [reply products/views.py]: Done.

File: products/views.py (hunk 3 above)
Comment (rinkishivani) [reply products/views.py]: removed.

File: products/views.py [hunk 4]
```diff
@@ -29,3 +36,111 @@ def home_page(request):
...
+
+@login_required
+def delete_brand(request, pk):
+    """ To delete a brand from model """
+
+    if not request.user.is_superuser:
+        return Http404
+
+    brand = Brand.objects.get(id=pk)
+    product_has_brand = BrandProduct.objects.filter(brand=pk)
```
Comment (snehals21) [inline products/views.py]: @rinkishivani if you are using `product_has_brand`  only to check whether brand has products or not then you can get brand's products by brand object using reverse relation and check exists(). 

reference: https://docs.djangoproject.com/en/5.0/ref/models/relations/
cc: @lalitvasoya

Comment (rinkishivani) [reply products/views.py]: Done

**Review by snehals21 — APPROVED** [review]

**Pull request merged**